src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -601,6 +640,7 @@ private bool IsEntryPoint(IMethodSymbol methodSymbol)
                 => methodSymbol.Name == WellKnownMemberNames.EntryPointMethodName &&
                    methodSymbol.IsStatic &&
                    (methodSymbol.ReturnsVoid ||
+                    methodSymbol.ReturnType.SpecialType == SpecialType.System_Int32 ||
                     methodSymbol.ReturnType.OriginalDefinition == _taskType ||
                     methodSymbol.ReturnType.OriginalDefinition == _genericTaskType);
```

Comment:
note: `==` with types isn't a great idea. (this is about the _taskType/_genericTaskType checks).

Reply:
Yep, will fix.